### ISSUE TITLE
Concatenating over many days

### ISSUE THREAD

**Issue by holmescharles** [spyking-circus/spyking-circus#146]
I'm not sure if this is a need for a new feature or if someone has a paradigm idea that could help, but here goes:

I am collecting data with a Graymatter SC32 (32 channel semi-chronic drive). From day to day, I advance the channels ~30 um, but if I have a cell on a channel, I will often leave it be for days. As such, from day to day, some channels move and some don't.

I would like to sort for spikes concatenating my data over days, but that will work for some channels and not for others. With 32 channels the times during which particular channels are stationary overlap a lot and one can see how this could get unruly very quickly.

The best solutions I can come up with are:
1. For each channel, concatenate all the data files during which the channel was stationary and sort only that channel. Downside: I'd have to do this for each channel, so that's 32 copies of the data.
2. For each channel, extract out only that channel's data and concatenate them. Downside: I lose the ability to remove common noise and whiten the data.

Does anyone have a better solution?
Or, would it be possible to add a feature in which you can indicate times during which the electrode was stationary and spikes can only be clustered with other spikes from those times (not with spikes from other times when the electrode was somewhere else)?

**Comment by yger** [comment]
I'm travelling, with a limited access to internet, and will only be back
end of the month.

I'll think about your problem meanwhile :-)

Le lun. 5 août 2019 à 22:45, Chuck Holmes <notifications@github.com> a
écrit :

**Comment by holmescharles** [comment]
Actually, data is cheap, so I don't mind the space of using multiple "copies" of the same data. What would be nice, though, would be the ability to include all channels in pre-processing (e.g., median-removal and whitening) but only look for spikes on a particular channel.

My go-to methodology for inlcuding/excluding channels has been via the probe file. However, if I only want to look at 1 channel, it will not do a median-removal with all the other channels (etc). Excluding channels in "dead channels" seems to have the same effect.

TLDR: Can there be 3 levels of channelds? Dead, live-but-don't-bother-looking-for-spikes, and live-look-for-spikes?

**Comment by yger** [comment]
I'm trying to catch up also with this issue.You already have the possibility, with the software, to provide a list of time intervals that you want to exclude from the analysis. We are calling them "dead_times". However, from what I understood, this will not be enough because you have different dead times on a per channel basis, while we assume that they are global. 3 levels of dead channels is feasible, but will it solve all your problems? If we were extending the notion of dead times to a per channel basis, will it be enough for you? This is easily feasible I think, and will allow you to exclude spikes found in given time intervals and on given electrodes

**Comment by holmescharles** [comment]
I think to make my analysis easier, I will just use global dead times. Technically, this is not *optimal*, but its much more practical. Thanks!

On Mon, Sep 16, 2019 at 7:01 AM Pierre Yger <notifications@github.com<mailto:notifications@github.com>> wrote:

I'm trying to catch up also with this issue.You already have the possibility, with the software, to provide a list of time intervals that you want to exclude from the analysis. We are calling them "dead_times". However, from what I understood, this will not be enough because you have different dead times on a per channel basis, while we assume that they are global. 3 levels of dead channels is feasible, but will it solve all your problems? If we were extending the notion of dead times to a per channel basis, will it be enough for you? This is easily feasible I think, and will allow you to exclude spikes found in given time intervals and on given electrodes

**Comment by yger** [comment]
Note also that the filtering will filter the signal even in the "dead" intervals. These intervals are simply portions of the signal you want to exclude to detect spikes (during clustering and fitting)
